src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1175 @@
+    config_value: when defined, make it part of the pattern (used when multiple lines
+                  with the same keyword are allowed)
+
+Return value:
+    A string containing the pattern to use to match the line in the file or undef
+    in case of an internal error (missing argument).
+
+=cut
+
+sub _buildLinePattern
+{
+    my $function_name = "_buildLinePattern";
+    my ($self, $config_param, $line_fmt, $config_value) = @_;
+
+    unless ($config_param) {
+        *$self->{LOG}->error("$function_name: 'config_param' argument missing (internal error)");
+        return undef;
+    }
+    unless (defined($line_fmt)) {
+        *$self->{LOG}->error("$function_name: 'line_fmt' argument missing (internal error)");
+        return undef;
+    }
+    if (defined($config_value)) {
+        *$self->{LOG}->debug(2, "$function_name: configuration value '$config_value' will be added to the pattern");
+        $config_value =~ s/\\/\\\\/g;
+        $config_value =~ s/([\-\+\?\.\*\[\]()\^\$])/\\$1/g;
+        $config_value =~ s/\s+/\\s+/g;
+    } else {
+        $config_value = "";
+    }
+
+    # config_param is generally a keyword and in this case it contains no whitespace.
+    # A special case is when config_param (the rule keyword) is used to match a line
+    # without specifying a rule: in this case it may contains whitespaces. Remove strict
+    # matching of them (match any type/number of whitespaces at the same position).
+    # Look at %trust_config_rules in ncm-dpmlfc Perl module for an example.
+    $config_param =~ s/\s+/\\s+/g;
+
+    my $config_param_pattern;
+    if ($line_fmt == LINE_FORMAT_SH_VAR) {
+        $config_param_pattern = "#?\\s*$config_param=" . $config_value;
+    } elsif ($line_fmt == LINE_FORMAT_ENV_VAR) {
+        $config_param_pattern = "#?\\s*export $config_param=" . $config_value;
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL_SETENV) {
+        $config_param_pattern = "#?\\s*setenv\\s+$config_param\\s*=\\s*" . $config_value;
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL_SET) {
+        $config_param_pattern = "#?\\s*set\\s+$config_param\\s*=\\s*" . $config_value;
+    } elsif ($line_fmt == LINE_FORMAT_KEY_VAL) {
+        $config_param_pattern = "#?\\s*$config_param";
+        # Avoid adding a whitespace requirement if there is no config_value
+        if ($config_value ne "") {
+            $config_param_pattern .= "\\s+" . $config_value;
+        }
+    } else {
+        *$self->{LOG}->error("$function_name: invalid line format ($line_fmt). Internal inconsistency.");
+        return undef;
+    }
+
+    return $config_param_pattern;
+}
+
+
+=pod
+
+=item _removeConfigLine
+
+This function comments out a configuration line matching the configuration parameter.
+Match operation takes into account the line format.
+
+Arguments :
+    config_param: parameter to update
+    line_fmt : line format (see LINE_FORMAT_xxx constants)
+
+Return value:
+    None or 1 in case of an internal error (missing argument)
```

Comment:
None -> undef; but returning 1 in case of failure, and undef in case of success is odd at best. why not stick with what you do above: return 1 on success and undef onfailure?